src/main/java/com/arpnetworking/metrics/mad/parsers/JsonToRecordParser.java

```diff
@@ -289,32 +249,62 @@ public Record parse(final byte[] data) throws ParsingException {
-            } else if (!Key.HOST_DIMENSION_KEY.equals(key)
-                    && !Key.SERVICE_DIMENSION_KEY.endsWith(key)
-                    && !Key.CLUSTER_DIMENSION_KEY.endsWith(key)) {
-                mappedAnnotations.put(key, value);
-            }
-        }
+        final ImmutableMap.Builder<String, String> annotationsBuilder = ImmutableMap.builder();
+        annotationsBuilder.putAll(annotations.getOtherAnnotations());
+        annotationsBuilder.put(PREFIXED_HOST_KEY, context.getHost());
+        final ImmutableMap<String, String> flatAnnotations = annotationsBuilder.build();
 
         return new DefaultRecord.Builder()
                 .setMetrics(variables.build())
                 .setTime(annotations.getEnd())
                 .setId(model.getId())
-                .setAnnotations(mappedAnnotations.build())
+                .setAnnotations(flatAnnotations)
+                .setDimensions(extractLegacyDimensions(flatAnnotations))
                 .build();
     }
 
+    // NOTE: Package private for testing
+    /* package private */com.arpnetworking.metrics.mad.model.Record parseV2gLogLine(final JsonNode jsonNode)
+            throws JsonProcessingException {
+
+        final Version2g model = OBJECT_MAPPER.treeToValue(jsonNode, Version2g.class);
+        final ImmutableMap.Builder<String, Metric> variables = ImmutableMap.builder();
+
+        putVariablesVersion2g(model.getTimers(), MetricType.TIMER, variables);
+        putVariablesVersion2g(model.getCounters(), MetricType.COUNTER, variables);
+        putVariablesVersion2g(model.getGauges(), MetricType.GAUGE, variables);
+
+        return new DefaultRecord.Builder()
+                .setMetrics(variables.build())
+                .setTime(model.getEnd())
+                .setId(model.getId())
+                .setAnnotations(ImmutableMap.copyOf(model.getAnnotations()))
+                .setDimensions(ImmutableMap.copyOf(model.getDimensions()))
+                .build();
+    }
+
+    private static ImmutableMap<String, String> extractLegacyDimensions(final Map<String, String> annotations) {
+
+        final ImmutableMap.Builder<String, String> defaultDimensions = ImmutableMap.builder();
+        for (final Map.Entry<String, String> dimensionName : LEGACY_DIMENSION_MAP.entrySet()) {
+            final String dimensionValue = annotations.get(dimensionName.getKey());
+            if (null != dimensionValue) {
+                defaultDimensions.put(dimensionName.getValue(), dimensionValue);
+            }
+        }
+
+        return defaultDimensions.build();
+    }
+
+    private static final String PREFIXED_HOST_KEY = "_host";
+    private static final String PREFIXED_SERVICE_KEY = "_service";
+    private static final String PREFIXED_CLUSTER_KEY = "_cluster";
+    private static final ImmutableMap<String, String> LEGACY_DIMENSION_MAP = ImmutableMap.of(
+            PREFIXED_HOST_KEY, Key.HOST_DIMENSION_KEY,
```

Comment:
The prefixed annotations are required by the file schema, in what context would you have a use for the non-prefixed value?